sccp/figures/figureS2.py

```diff
@@ -28,31 +26,32 @@ def makeFigure():
     # Add subplot labels
     subplotLabel(ax)
 
+    rank = 40
+    cmp = 13
+
     # Import of data
-    data, obs = load_lupus_data(every_n=10)  # don't need to get patient color mappings
-    rank = 5
-    cmp = 4
+    _, obs= load_lupus_data()
+    
+    broad_type = obs["cell_type_broad"].reset_index(drop=True)
+    lympho_type = obs["cell_type_lympho"].reset_index(drop=True)
+
+    # replace NaN with string
+    lympho_type = lympho_type.cat.add_categories('other').fillna('other')
 
-    cell_types = obs["cell_type_broad"].reset_index(drop=True)
+    _, factors, projs, = openPf2(rank = rank, dataName = 'lupus', optProjs=True)
 
-    # run pf2
-    _, factors, projs, _ = parafac2_nd(
-        data,
-        rank=rank,
-        n_iter_max=10,
-        random_state=1,
-    )
 
-    projs = np.concatenate(projs, axis=0)
     # UMAP dimension reduction
     pf2Points = umap.UMAP(random_state=1, verbose=True).fit(projs)
 
-    # PCA dimension reduction
-    pc = PCA(n_components=rank)
-    pcaPoints = pc.fit_transform(data.unfold())
-    pcaPoints = umap.UMAP(random_state=1).fit(pcaPoints)
+    # IF RUNNING MANY TIMES LOCALLY: CAN PICKLE DUMP AND LOAD
+    #f_name = 'pf2Points_40comp.sav'
+    #pickle.dump(pf2Points, open(f_name, 'wb')) # do this first time; then comment and run load subsequently
+    #pf2Points = pickle.load((open(f_name, 'rb')))
+
```

Comment:
could actually add this file to opt or smth if we want this to build faster

Reply:
@pascoesean Can you make this into a function in common.py so it can be used for other functions? How big is the file if you save it?

Reply:
its 1.4 GB :0 there might be a more compressed way to save something like this? But mostly I was copying code I found online and it worked